src/client/options/mod.rs

```diff
@@ -834,38 +977,39 @@ pub struct DriverInfo {
     pub platform: Option<String>,
 }
 
-impl From<ClientOptionsParser> for ClientOptions {
-    fn from(parser: ClientOptionsParser) -> Self {
+impl From<ConnectionString> for ClientOptions {
```

Comment:
I think we actually don't want this implemented (and didn't before either really), since it bypasses the SRV resolution. Before it was somewhat okay since `ClientOptionsParser` was `pub(crate)`, but now that `ConnectionString` is public we'll probably need to remove this.

Instead, we can just update the parts in `ClientOptions::parse` that use this to first resolve the hosts (either by just getting them directly from the `ConnectionString` or by doing SRV lookup) and then basically doing the same thing that's done here but inline.

Reply:
Ah, didn't consider visibility - it's a pity there's no non-awkward way to have a non-pub `impl`.  Updated, although I had to preserve the struct-translation as its own method so it could be shared with `parse_without_srv_resolution`.